core/dbt/parser/read_files.py

```diff
@@ -153,104 +172,243 @@ def generate_dbt_ignore_spec(project_root):
-        project,
-        files,
-        project.seed_paths,
-        [".csv"],
-        ParseFileType.Seed,
-        saved_files,
-        dbt_ignore_spec,
-    )
+    def read_files(self):
+        # Copy the base file information from the existing manifest.
+        # We will do deletions, adds, changes from the file_diff to emulate
+        # a complete read of the project file system.
+        for file_id, source_file in self.saved_files.items():
+            if isinstance(source_file, SchemaSourceFile):
+                file_cls = SchemaSourceFile
+            else:
+                file_cls = SourceFile
+            new_source_file = file_cls(
+                path=source_file.path,
+                checksum=source_file.checksum,
+                project_name=source_file.project_name,
+                parse_file_type=source_file.parse_file_type,
+                contents=source_file.contents,
+            )
+            self.files[file_id] = new_source_file
 
-    project_files["DocumentationParser"] = read_files_for_parser(
-        project,
-        files,
-        project.docs_paths,
-        [".md"],
-        ParseFileType.Documentation,
-        saved_files,
-        dbt_ignore_spec,
-    )
+        # Now that we have a copy of the files, remove deleted files
+        # For now, we assume that all files are in the root_project, until
+        # we've determined whether project name will be provided or deduced
+        # from the directory.
+        for input_file_path in self.file_diff.deleted:
+            project_name = self.get_project_name(input_file_path)
+            file_id = f"{project_name}://{input_file_path}"
+            if file_id in self.files:
+                self.files.pop(file_id)
+            else:
+                fire_event(InputFileDiffError(category="deleted file not found", file_id=file_id))
+
+        # Now we do the changes
+        for input_file in self.file_diff.changed:
+            project_name = self.get_project_name(input_file.path)
+            file_id = f"{project_name}://{input_file.path}"
+            if file_id in self.files:
+                # Get the existing source_file object and update the contents and mod time
+                source_file = self.files[file_id]
+                source_file.contents = input_file.content
+                source_file.checksum = FileHash.from_contents(input_file.content)
+                source_file.path.modification_time = input_file.modification_time
+                # Handle creation of dictionary version of schema file content
+                if isinstance(source_file, SchemaSourceFile) and source_file.contents:
+                    dfy = yaml_from_file(source_file)
+                    if dfy:
+                        validate_yaml(source_file.path.original_file_path, dfy)
+                        source_file.dfy = dfy
+                    # TODO: ensure we have a file object even for empty files, such as schema files
```

Comment:
What if we leave the empty part to where we figure out what are the file diffs? Would that leads to simpler code on the parsing side?
I think it is reasonable to thing the input to parsing could be a manifest + file diff in the following format
```
{
  "updated":{
    "file1": "full content of file1",
    "file2": "full content of file2",
    ...
  },
  "added":{
    "file3": "full content of file3",
    "file4": "full content of file4",
    ...
  },
 "deleted": ["file4", "file5"],
}
```
`updated` and `added` could also be merged into one dictionary.

From dat-core's perspective an empty file would be the same as that file doesn't exist

Reply:
This is the format that we're using now -- are you suggesting something different?

The empty file thing only applies to schema files, and we can't depend on it being simply an empty string -- it might have comments or whitespace, etc. I've changed not storing a schema file if it doesn't reduce to a real dictionary.

Reply:
I think I missed the fact that this is just for schema files. And the comments/whitespace are good call out!